db/database.go

```diff
@@ -945,24 +948,26 @@ func (db *Database) UpdateAllDocChannels(doCurrentDocs bool, doImportDocs bool)
 		}
 		var err error
 		if db.UseXattrs() {
-			err = db.Bucket.WriteUpdateWithXattr(key, KSyncXattrName, 0, func(currentValue []byte, currentXattr []byte, cas uint64) (raw []byte, rawXattr []byte, err error) {
+			err = db.Bucket.WriteUpdateWithXattr(key, KSyncXattrName, 0, func(currentValue []byte, currentXattr []byte, cas uint64) (raw []byte, rawXattr []byte, deleteDoc bool, err error) {
+				// There's no scenario where a doc should from non-deleted to deleted during UpdateAllDocChannels processing, so deleteDoc is always returned as false.
```

Comment:
The nature of UpdateAllDocChannels (rerunning the sync function against existing docs) means that documents aren't going to change from non-deleted to deleted.  I previously had an explicit 'deleteDoc=false' here, but took it out since it wasn't necessary.  I could add it back in if you think it's needed for clarity.